src/lib/FIFO/FIFO.cpp

```diff
@@ -42,7 +42,7 @@ FIFO::~FIFO()
 
 void ICACHE_RAM_ATTR FIFO::push(uint8_t data)
 {
-    if (numElements == FIFO_SIZE)
+    if (numElements == (FIFO_SIZE-1))
```

Comment:
Hmmm, would it be better to define FIFO_SIZE as 255 with a comment,
or add some preprocessor code to check that it fits in a `uint8_t`
```
#if FIFO_SIZE > 255
#error FIFO_SIZE must be less that 256
#endif
```
I'd put the above where numElements is defined.

Reply:
I've added a compile time check to make sure FIFO_SIZE does not exceed the capacity of numElements and changed the code to what you've suggested, so size will actually be the size, not size+1. Good note!